File: devcamp2018/app/src/main/java/com/psato/devcamp/data/repository/show/ShowRepository.java
package com.psato.devcamp.data.repository.show;

import com.psato.devcamp.data.entity.ShowInfo;
import com.psato.devcamp.data.entity.ShowRating;

import java.util.List;

import io.reactivex.Single;

/**
 * Created by psato on 29/06/16.
 */

public interface ShowRepository {
    Single<List<ShowInfo>> searchShow(String query);

    Single<ShowRating> showRating(String id);
}

File: devcamp2018/app/src/main/java/com/psato/devcamp/data/repository/show/ShowRepository.kt
package com.psato.devcamp.data.repository.show

import com.psato.devcamp.data.entity.Rating
import com.psato.devcamp.data.entity.ShowInfo
import java.io.IOException

/**
 * Created by psato on 29/06/16.
 */

interface ShowRepository {
    @Throws(IOException::class)
    suspend fun searchShow(query: String): List<ShowInfo>

    @Throws(IOException::class)
    suspend fun showRating(id: String): Rating
}
